### PR TITLE
chore: simplifying fuel error implementation

### PR DESCRIPTION
**Pull request by arboleya**
Re-writing of #1193 following internal conversations.

 1. Simplify `expectToThrowFuelError` test utility
 1. Re-write all tests accordingly
 1. Improve package README

Bonus:
 1. Adding new `eslint` rule to prevent floating promises ([here](https://github.com/FuelLabs/fuels-ts/blob/83e954f3ddf7acb6e5e96d68ed107d377584c302/.eslintrc.js#L54))
 1. Fixing all floating promises

### CODE REVIEW

File: apps/docs/scripts/typedoc-postbuild.ts [hunk 1]
```diff
@@ -174,4 +174,4 @@ const main = async () => {
   recreateInternalLinks();
 };
 
-main();
+main().catch(process.stderr.write);
```
Comment (danielbate) [inline apps/docs/scripts/typedoc-postbuild.ts:177]: What was this in aid of?

Comment (arboleya) [reply apps/docs/scripts/typedoc-postbuild.ts:177]: After turning on [this](https://github.com/FuelLabs/fuels-ts/pull/1228/commits/8121acbc5320024e2d67ec7f2ea3f1783d12021a) `eslit` rule, I had to fix all unhandled promises, and this was one of them.

You can see it all in this commit. https://github.com/FuelLabs/fuels-ts/pull/1228/commits/83e954f3ddf7acb6e5e96d68ed107d377584c302

**Review by Torres-ssf — APPROVED** [review]

File: packages/errors/src/test-utils/expect-to-throw-fuel-error.ts [hunk 2]
```diff
@@ -1,56 +1,37 @@
...
+  const codes = Object.values(ErrorCode);
+
+  if (!thrownError) {
+    throw new Error("Passed-in lambda didn't throw.");
   }
 
-  if (!errorCodes.includes(expectedError.code)) {
-    throw new Error(`Expected error code '${expectedError.code}' is not a valid FuelError code.`);
+  if (!thrownError.code) {
+    throw new Error('Thrown error must contain a code.');
```
[the remark this answers is not in the document]
Comment (arboleya) [reply packages/errors/src/test-utils/expect-to-throw-fuel-error.ts:29]: There are cases where one will provoke the error on purpose.

Wouldn't logging stuff inside the utility pollute the test's output?

Users can also investigate problems by executing the lambda directly.

What do you think?

**Review by Torres-ssf — APPROVED** [review]

**Review by danielbate — DISMISSED** [review]

File: packages/errors/src/test-utils/expect-to-throw-fuel-error.ts (hunk 2 above)
[the remark this answers is not in the document]
Comment (arboleya) [reply packages/errors/src/test-utils/expect-to-throw-fuel-error.ts:29]: Gotcha, let me see what I can do.

**Review by Torres-ssf — DISMISSED** [review]

File: packages/errors/README.md [hunk 3]
```diff
@@ -45,6 +46,32 @@ export function multipleImports() {
...
+
+You can use the `expectToThrowFuelError` test utility when asserting errors.
+
+```ts
+// can also be imported from `fuels` when using the umbrella package
+import { expectToThrowFuelError } from "@fuel-ts/errors/test-utils";
+
+import { myFn } from "...";
+
+describe('this and that'm () => {
```
Comment (danielbate) [inline packages/errors/README.md:59]:
```suggestion
describe('this and that' () => {
```

Comment (danielbate) [reply packages/errors/README.md:59]: Another case for test backed snippets 🤔  Potentially overcomplicating a simple issue but we use a markdown generator to pass tests to files.

**Review by danielbate — CHANGES_REQUESTED** [review]

File: packages/errors/README.md (hunk 3 above)
Comment (arboleya) [reply packages/errors/README.md]: That's a good point, although we'd need to develop a routine to validate the READMEs.

I'm resolving this for now as it stands outside the scope of this PR.

**Review by Torres-ssf — APPROVED** [review]

**Review by danielbate — APPROVED** [review]

**Pull request merged**